packages/x-components/src/x-modules/facets-next/components/filters/__tests__/hierarchical-filter.spec.ts

```diff
@@ -0,0 +1,489 @@
+import { HierarchicalFilter as HierarchicalFilterModel } from '@empathyco/x-types-next';
+import { mount, Wrapper } from '@vue/test-utils';
+import Vue from 'vue';
+import { createNextHierarchicalFacetStub } from '../../../../../__stubs__/facets-stubs.factory';
+import { createNextHierarchicalFilter } from '../../../../../__stubs__/filters-stubs.factory';
+import { getDataTestSelector, installNewXPlugin } from '../../../../../__tests__/utils';
+import { getXComponentXModuleName, isXComponent } from '../../../../../components';
+import { XPlugin } from '../../../../../plugins/x-plugin';
+import { arrayToObject } from '../../../../../utils/array';
+import { XEvent } from '../../../../../wiring/events.types';
+import { facetsNextXModule } from '../../../x-module';
+import { resetXFacetsStateWith } from '../../__tests__/utils';
+import HierarchicalFilter from '../hierarchical-filter.vue';
+
+function renderHierarchicalFilter({
+  template = `<HierarchicalFilter :filter="filter" />`,
+  filter = createNextHierarchicalFilter('category', 'clothing'),
+  filters = [filter]
+}: HierarchicalFilterOptions = {}): HierarchicalFilterAPI {
+  const [, localVue] = installNewXPlugin({ initialXModules: [facetsNextXModule] });
+  const emit = jest.spyOn(XPlugin.bus, 'emit');
+  const store = XPlugin.store;
+  resetXFacetsStateWith(store, { filters: arrayToObject(filters, 'id') });
```

Comment:
This is failing in the tests.

Reply:
Yes, the `FacetNextState` needs more properties like `query`, `groups` or `facets`

```
export interface FacetsNextState {
  filters: Record<Filter['id'], Filter>;
  groups: Record<Facet['id'], GroupId>;
  query: string;
  facets: Record<Facet['id'], Omit<Facet, 'filters'>>;
}
```